binding/button_test.go

```diff
@@ -0,0 +1,63 @@
+package binding_test
+
+import (
+	"github.com/stretchr/testify/assert"
+	"testing"
+	"time"
+
+	"fyne.io/fyne/binding"
+	"fyne.io/fyne/test"
+	"fyne.io/fyne/theme"
+	"fyne.io/fyne/widget"
+)
+
+func TestBindButtonTapped_Binding(t *testing.T) {
+	a := test.NewApp()
+	defer a.Quit()
+	tapped := false
+	button := widget.NewButton("button", func() {
+		tapped = true
+	})
+	data := &binding.BoolBinding{}
+	binding.BindButtonTapped(button, data)
+	data.Set(true)
+	time.Sleep(time.Second)
```

Comment:
This is a substantial delay - what is it waiting for?
Adding timing in tests is usually a warning flag ;)

Reply:
A very good suggestion Andy - I changed to a chan bool and select w/ timeout, and test execution dropped from 18.280s to 0.388s